src/main/java/org/elasticsearch/percolator/PercolatorService.java

```diff
@@ -843,15 +847,29 @@ private InternalAggregations reduceAggregations(List<PercolateShardResponse> sha
             return null;
         }
 
+        InternalAggregations aggregations;
         if (shardResults.size() == 1) {
-            return shardResults.get(0).aggregations();
-        }
-
+            aggregations = shardResults.get(0).aggregations();
+        } else {
```

Comment:
Ah, yes. Sorry I misunderstood. Yes we should be always calling reduce(). I'll just keep the else part as you suggest